From f3aaf4a35a6cadfc260149c332d5519e418135c6 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Mon, 8 Nov 2010 10:49:32 +0100
Subject: [PATCH] Fix popping of vars in xsltCompilerNodePop
To: libvir-list@redhat.com

Fix an off by one bug in Var pop-up in the compiler

Signed-off-by: Daniel Veillard <veillard@redhat.com>
---
 libxslt/xslt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libxslt/xslt.c b/libxslt/xslt.c
index 55f505b..28c8c59 100644
--- a/libxslt/xslt.c
+++ b/libxslt/xslt.c
@@ -2432,13 +2432,13 @@ xsltCompilerNodePop(xsltCompilerCtxtPtr cctxt, xmlNodePtr node)
 	"xsltCompilerNodePop: Depth mismatch.\n");
 	goto mismatch;
     }
+    cctxt->depth--;
     /*
     * Pop information of variables.
     */
     if ((cctxt->ivar) && (cctxt->ivar->depth > cctxt->depth))
 	xsltCompilerVarInfoPop(cctxt);
 
-    cctxt->depth--;
     cctxt->inode = cctxt->inode->prev;
     if (cctxt->inode != NULL)
 	cctxt->inode->curChildType = 0;
-- 
1.7.11.4

